### DISCUSSION TITLE
Action on import depending of choice made during prompt

### DISCUSSION THREAD

**Discussion by Vitriia** [beetbox/beets#4461] [Ideas]
Hi everyone !

I have a question/new feature idea (it can be a plugin).

It is possible to do actions on import depending of the choice made during a beets prompt ?
For example filling concerned tracks with a tag `beets_import` with `as-in` value when you have chose As-in option during import ?

I have chosen tags because they can be utilized in multiple ways in Beets : you can retry imports on tag-matching tracks later; you can import/move those tag-matching to a specific folder etc etc.
Like this when importing your music you will be able to have your indexed tracks in a folder, as-in in another, skipped in a third. With the possibility to extend it to duplicate (on import having duplicates moved in a specific folder).

I suppose this idea have been shared before but i haven't been able to find another post talking of it here, on the wiki or with a search engine.
Sorry for my english i'm not native

Have a great day :)

**Comment by sampsyo** [comment]
This does seem quite useful; good idea all around!

I think you may be able to accomplish something like this with the `hook` plugin:
https://beets.readthedocs.io/en/stable/plugins/hook.html

In particular, take a look at the `import_task_choice` event. That event receives an object called `task` that contains information about what the user just selected. You should be able to use `task.choice_flag` to make decisions based on what that choice was.

**Reply by Vitriia** [reply]
The problem is : 
I don't know how to program so for me this part of the doc is largely extraterrestrial language ^^"
